### PR TITLE
wire screen 4 to data, add service provider contact objects to model

### PR DESCRIPTION
**Pull request by bethbrains**
# Critical Changes

# Changes

# Issues Closed

# New Metadata

# Deleted Metadata

# Definition of Done
  Refer to [Asteroids DoD document](https://salesforce.quip.com/iq2mAy4i62oM) to see any additional details for the items below
- [ ] Place holder data is incorporated (Refer to [Case Management Sample Data document](https://quip.com/cbFcAPJF8t0z))
- [ ] Base axe-core JEST test included for any new LWC (No critical violations)
- [ ] CRUD/FLS is enforced in Apex
- [ ] Update & Test permission sets to account for new features and CRUD/FLS
- [ ] All modified classes are unit tested (Apex) at 100% coverage
- [ ] UX approval or UX not necessary
- [ ] PR contains draft release notes
- [ ] Attach work item to the pull request with [Lurch](https://salesforce.quip.com/50ZRA5LEWVzH) `**Lurch:attach W-000000` or `**Lurch:add`
- [ ] All **acceptance criteria** have been met
    - [ ] Developer
    - [ ] Code Reviewer
    - [ ] QA
- [ ] QE story level testing completed

### CODE REVIEW

**Review by lbeller — APPROVED** [review]

File: force-app/main/default/objects/ProgramEngagement__c/fieldSets/SessionParticipantView.fieldSet-meta.xml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        <field>Contact__r.Email</field>
+        <isFieldManaged>false</isFieldManaged>
+        <isRequired>false</isRequired>
+    </displayedFields>
+    <displayedFields>
+        <field>Stage__c</field>
+        <isFieldManaged>false</isFieldManaged>
+        <isRequired>true</isRequired>
+    </displayedFields>
+    <label>CreateProgramEngagement</label>
```
Comment (bmuniswamy) [inline force-app/main/default/objects/ProgramEngagement__c/fieldSets/SessionParticipantView.fieldSet-meta.xml:70]: Can we have the name of the label and apiName be the same.

**Review by bmuniswamy — CHANGES_REQUESTED** [review]
I have a couple of questions.  Noticed a couple of things in the UI:

- So on the last screen if there are no participants or sessions being created we may want to add a message saying there is nothing selected. We can move this to the UI refinements story.
- Noticed that screen 3 UI does not go off of the fieldset. We can move that to the UI refinements story.
- This is a bug that I noticed and we may want to address it here: So when I create a new schedule and on screen 2 remove all the sessions and go to screen 3 and click next on screen 3 the wizard breaks silently. When we click back it has a weird behavior. 

If we are moving the first 2 points to the UI story we may want to let the QE's know since they may flag that as an issue.

File: force-app/main/default/lwc/reviewSessions/reviewSessions.js [hunk 2]
```diff
@@ -40,18 +38,23 @@ export default class ReviewSessions extends LightningElement {
         return this._serviceSessions;
     }
 
+    @api
+    get serviceSchedule() {
+        // now contains full contact records for Service Provider lookups
```
[the remark this answers is not in the document]
Comment (bethbrains) [reply force-app/main/default/lwc/reviewSessions/reviewSessions.js:43]: i thought it was useful to explain _why_ we had to get it again.

File: force-app/main/default/classes/ServiceScheduleService.cls [hunk 3]
```diff
@@ -27,7 +29,41 @@ public with sharing class ServiceScheduleService {
...
+        setServiceProviders(model.serviceSchedule);
+        return model;
+    }
+
+    private void setServiceProviders(ServiceSchedule__c schedule) {
+        Map<Id, Contact> providerById = new Map<Id, Contact>();
+
+        providerById.put(schedule.PrimaryServiceProvider__c, null);
+        providerById.put(schedule.OtherServiceProvider__c, null);
+        providerById.remove(null);
```
[the remark this answers is not in the document]
Comment (jjbennett) [reply force-app/main/default/classes/ServiceScheduleService.cls:43]: I was trying to reduce the redundant if checks but yea it's not readable so I will change it.

File: force-app/main/default/classes/ServiceScheduleCreatorController_TEST.cls [hunk 4]
```diff
@@ -42,22 +42,20 @@ public with sharing class ServiceScheduleCreatorController_TEST {
...
-            schedule
-        );
+        ServiceScheduleModel actual = ServiceScheduleCreatorController.updateModel(model);
         Test.stopTest();
 
-        serviceStub.assertCalledWith(methodName, ServiceSchedule__c.class, schedule);
+        serviceStub.assertCalledWith(methodName, ServiceScheduleModel.class, model);
         System.assertEquals(
             expected,
             actual,
```
[the remark this answers is not in the document]
Comment (jjbennett) [reply force-app/main/default/classes/ServiceScheduleCreatorController_TEST.cls:61]: The stub will provide a message here.

File: force-app/main/default/objects/ProgramEngagement__c/fieldSets/SessionParticipantView.fieldSet-meta.xml (hunk 1 above)
Comment (jjbennett) [reply force-app/main/default/objects/ProgramEngagement__c/fieldSets/SessionParticipantView.fieldSet-meta.xml]: Nice catch!

**Review by bmuniswamy — APPROVED** [review]
This is amazing work Beth and Jenny !!!

**Pull request merged**